### ISSUE TITLE
Metamodel fails to parse array default value

### ISSUE THREAD

**Issue by rbuehlma** [jOOQ/jOOQ#11364]
### Expected behavior

Since it is probably hard to parse all possible default values, it should be possible to retrieve a raw default value in case Jooq was not able to parse the actual value.

### Actual behavior

Jooq throws the below error and "defaultValue" is null in the meta model.
```
17:29:39.800 [main] WARN  org.jooq.impl.MetaImpl - Default value            : Could not load default value: '{}'::character varying[] for type: varchar(10) array /* [Ljava.lang.String; */
org.jooq.exception.DataTypeException: Cannot convert from '{}'::character varying[] (class java.lang.String) to class [Ljava.lang.String;
	at org.jooq.tools.Convert$ConvertAll.fail(Convert.java:1282) ~[jooq-3.14.7.jar:?]
	at org.jooq.tools.Convert$ConvertAll.from(Convert.java:1181) ~[jooq-3.14.7.jar:?]
	at org.jooq.tools.Convert.convert0(Convert.java:380) ~[jooq-3.14.7.jar:?]
	at org.jooq.tools.Convert.convert(Convert.java:453) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.AbstractDataType.convert(AbstractDataType.java:534) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.DefaultDataType.convert(DefaultDataType.java:92) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.DSL.val(DSL.java:24409) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.DSL.inline(DSL.java:23927) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.MetaImpl$MetaTable.init(MetaImpl.java:904) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.MetaImpl$MetaTable.<init>(MetaImpl.java:556) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.MetaImpl$MetaSchema.getTables(MetaImpl.java:417) ~[jooq-3.14.7.jar:?]
	at org.jooq.impl.SchemaImpl.getTable(SchemaImpl.java:136) ~[jooq-3.14.7.jar:?]
```

### Steps to reproduce the problem
```
    @Test
    public void testCreateTable() throws Exception {
        Connection conn = getSourceDbConnection();
        DSLContext dsl = DSL.using(conn);
        dsl.execute("create table test_table (array_col varchar(10)[] not null default '{}')");
        org.jooq.Table<?> testTable = dsl.meta().getSchemas("test-source").get(0).getTable("test_table");
        Field field = testTable.field(0);
        DataType dataType = field.getDataType(dsl.configuration());
        assertThat(dataType.defaultValue(), Matchers.notNullValue());
    }
```
### Versions

- JOOQ: 3.14.7
- Java: 11
- Database (include vendor): Postgresql 10.7
- OS: Linux
- JDBC Driver (include name if inofficial driver): org.postgresql (42.2.5)

**Comment by rbuehlma** [comment]
Something similar happens here:

```
    @Test
    public void testCreateTable() throws Exception {
        Connection conn = getSourceDbConnection();
        DSLContext dsl = DSL.using(conn);
        //dsl.execute("create table test_table (array_col varchar(10)[] not null default '{}')");
        dsl.execute("create table test_table (id bigint default char_length('defaultValue'))");
        org.jooq.Table<?> testTable = dsl.meta().getSchemas("test-source").get(0).getTable("test_table");
        Field field = testTable.field(0);
        DataType dataType = field.getDataType(dsl.configuration());
        assertThat(dataType.defaultValue(), Matchers.notNullValue());
        assertThat(dataType.defaultValue().toString(), Matchers.not("null"));
    }
 ```
 
 The default value is "null" (the string, not a null reference)

**Comment by lukaseder** [comment]
Thanks a lot for your patience.

I agree we cannot parse all possible expressions. On the other hand, version 3.14.7 doesn't parse things yet, I believe? We've fixed a few issues in 3.15.0 only, see

https://github.com/jOOQ/jOOQ/blob/ad21db797afe534c264686c8be54e898938caabe/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L1073-L1091

In particular, this has been implemented now: https://github.com/jOOQ/jOOQ/issues/8469

Without the strategic decision of parsing default expression (or at least attempting to), and using plain SQL only when parsing fails, we can't really do much about these unknown expression syntaxes. One option is to backport #8469, but I fear that could introduce unforeseen regressions in a late patch release 3.14.8 where customers expect jOOQ 3.14.x to stabilise, not introduce significant new features.

Both of your examples work just fine for me in 3.15.0-SNAPSHOT. I'm thus closing this as a duplicate of #8469.
